src/views/admin/registerProducts.js

```diff
@@ -33,22 +33,9 @@ class RegisterProducts extends React.Component {
     
   submitHandler = e => {
     e.preventDefault();
-    const config = {
-      headers: {
-        'Accept': 'application/json',
-        'Authorization': localStorage.getItem('serverest/userToken'),
-      }
-    };
 
      if (this.state.imagem == '' || null || undefined) {
-      axios
-      .post('https://serverest.dev/produtos',
-        {
-          nome: this.state.name,
-          preco: this.state.price,
-          descricao: this.state.description,
-          quantidade: this.state.quantity,
-      }, config)
+      registerProduct(this.state.name, this.state.price, this.state.description, this.state.quantity,)
```

Comment:
```suggestion
      registerProduct(this.state)
```
Como `this.state` é um objeto e já possui `name`, `price`, `description` e `quantity`, acredito que assim irá funcionar. Só não deve funcionar se tiver mais alguma propriedade além das listadas.
**Vai ser preciso fazer esse mesmo ajuste em todos os locais que chamam `registerProduct`.**

Caso a sugestão acima não funcione, tente a seguinte:

```suggestion
      registerProduct({ 
        name: this.state.name,
        price: this.state.price,
        description: this.state.description,
        quantity: this.state.quantity
      })
```

_Obsv.: Não sei se a indentação está correta, vai ser preciso verificar na IDE._